gencode/gencode.ml

```diff
@@ -0,0 +1,266 @@
+type emoji =
+  { code_point : string
+  ; emoji : string
+  ; description : string
+  ; name : string
+  ; category : string
+  ; sub_category : string
+  }
+
+let string_escape_hex =
+  let fst_code = Char.code '0' in
+  let snd_code = Char.code 'a' - 10 in
+  let nibble_to_hex_char n =
+    if n < 0 || n >= 16 then
+      invalid_arg "nibbles must be within the range 0x0 and 0xf"
+    else if n < 10 then Char.chr (fst_code + n)
+    else Char.chr (snd_code + n)
+  in
+  fun s ->
+    let buf = Buffer.create (String.length s) in
```

Comment:
`String.length s * 4` ?

Reply:
?

Reply:
For each character in `s` we're adding 4 character in the buffer (look at the code right after) so we can directly initialize it to the correct size.